### PR TITLE
webrtc: add SRD test with two m-lines

### PR DESCRIPTION
**Pull request by fippo**
adds a setRemoteDescription test with two m-lines of the same kind.

@juberti @adamroach @ekr can you please check that this SDP is what JSEP has for what has been known as "unified plan"?

### CODE REVIEW

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/230238603)

# Firefox (nightly channel) #
Testing web-platform-tests at revision 75097788829161ff2753af33fd713ad160f9c3cb
Using browser at version BuildID 20170508100218; SourceStamp 1fda52a1f3b81cf1a821155998dca637bb64e3d9
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5858/webrtc/RTCPeerConnection-setRemoteDescription.html">/webrtc/RTCPeerConnection-setRemoteDescription.html</a></summary>


|                                             Subtest                                              | Results |              Messages              |
|--------------------------------------------------------------------------------------------------|---------|------------------------------------|
|                                                                                                  | OK      |                                    |
| `Triggers ontrack when called with a remote description and the MSID of the stream is is parsed` | PASS    |                                    |
| `Malformed SDP description should be rejected with RTCError`                                     | FAIL    | `RTCError is not defined`          |
| `Invalid SDP type should be rejected with TypeError`                                             | PASS    |                                    |
| `SDP type that is invalid for current signaling state should be rejected with InvalidStateError` | FAIL    | `InvalidStateError is not defined` |
| `Triggers ontrack twice when called with a remote description than contains two audio m-lines`   | PASS    |                                    |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/230238609)

# Chrome (unstable channel) #
Testing web-platform-tests at revision 75097788829161ff2753af33fd713ad160f9c3cb

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/233099911)

# Firefox (nightly) #
Testing web-platform-tests at revision c4bae14bcae87e92b3b4da2b2d4dea0bc29b3b0c
Using browser at version BuildID 20170516192431; SourceStamp b8e9b674033bcd1f3a4c59b9d0ee7619c1a17cc5
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5858/webrtc/RTCPeerConnection-setRemoteDescription.html">/webrtc/RTCPeerConnection-setRemoteDescription.html</a></summary>


|                                             Subtest                                              | Results |              Messages              |
|--------------------------------------------------------------------------------------------------|---------|------------------------------------|
|                                                                                                  | OK      |                                    |
| `Triggers ontrack when called with a remote description and the MSID of the stream is is parsed` | PASS    |                                    |
| `Malformed SDP description should be rejected with RTCError`                                     | FAIL    | `RTCError is not defined`          |
| `Invalid SDP type should be rejected with TypeError`                                             | PASS    |                                    |
| `SDP type that is invalid for current signaling state should be rejected with InvalidStateError` | FAIL    | `InvalidStateError is not defined` |
| `Triggers ontrack twice when called with a remote description than contains two audio m-lines`   | PASS    |                                    |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/233099912)

# Sauce (safari) #
Testing web-platform-tests at revision c4bae14bcae87e92b3b4da2b2d4dea0bc29b3b0c
Using browser at version 10.0
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5858/webrtc/RTCPeerConnection-setRemoteDescription.html">/webrtc/RTCPeerConnection-setRemoteDescription.html</a></summary>


|                                             Subtest                                              | Results |                 Messages                 |
|--------------------------------------------------------------------------------------------------|---------|------------------------------------------|
|                                                                                                  | OK      |                                          |
| `Triggers ontrack when called with a remote description and the MSID of the stream is is parsed` | FAIL    | `Can't find variable: RTCPeerConnection` |
| `Malformed SDP description should be rejected with RTCError`                                     | FAIL    | `Can't find variable: RTCPeerConnection` |
| `Invalid SDP type should be rejected with TypeError`                                             | FAIL    | `Can't find variable: RTCPeerConnection` |
| `SDP type that is invalid for current signaling state should be rejected with InvalidStateError` | FAIL    | `Can't find variable: RTCPeerConnection` |
| `Triggers ontrack twice when called with a remote description than contains two audio m-lines`   | FAIL    | `Can't find variable: RTCPeerConnection` |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/233099913)

# Sauce (MicrosoftEdge) #
Testing web-platform-tests at revision c4bae14bcae87e92b3b4da2b2d4dea0bc29b3b0c
Using browser at version 14.14393
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5858/webrtc/RTCPeerConnection-setRemoteDescription.html">/webrtc/RTCPeerConnection-setRemoteDescription.html</a></summary>


|                                             Subtest                                              | Results |              Messages              |
|--------------------------------------------------------------------------------------------------|---------|------------------------------------|
|                                                                                                  | OK      |                                    |
| `Triggers ontrack when called with a remote description and the MSID of the stream is is parsed` | FAIL    | `'RTCPeerConnection' is undefined` |
| `Malformed SDP description should be rejected with RTCError`                                     | FAIL    | `'RTCPeerConnection' is undefined` |
| `Invalid SDP type should be rejected with TypeError`                                             | FAIL    | `'RTCPeerConnection' is undefined` |
| `SDP type that is invalid for current signaling state should be rejected with InvalidStateError` | FAIL    | `'RTCPeerConnection' is undefined` |
| `Triggers ontrack twice when called with a remote description than contains two audio m-lines`   | FAIL    | `'RTCPeerConnection' is undefined` |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/233099919)

# Chrome (unstable) #
Testing web-platform-tests at revision c4bae14bcae87e92b3b4da2b2d4dea0bc29b3b0c

File: webrtc/RTCPeerConnection-setRemoteDescription.html [hunk 1]
```diff
@@ -82,6 +82,57 @@
...
+        'a=ice-ufrag:someufrag\r\n' +
+        'a=ice-pwd:somelongpwdwithenoughrandomness\r\n' +
+        'a=fingerprint:sha-256 8C:71:B3:8D:A5:38:FD:8F:A4:2E:A2:65:6C:86:52:BC:E0:6E:94:F2:9F:7C:4D:B5:DF:AF:AA:6F:44:90:8D:F4\r\n' +
+        'a=setup:actpass\r\n' +
+        'a=rtcp-mux\r\n' +
+        'a=mid:mid2\r\n' +
+        'a=sendonly\r\n' +
+        'a=rtpmap:111 opus/48000/2\r\n' +
+        'a=msid:stream2 track2\r\n' +
+        'a=ssrc:1001 cname:some\r\n';
```
Comment (nils-ohlmeier) [inline webrtc/RTCPeerConnection-setRemoteDescription.html]: You should probably not re-use the SSRC from track1 here.

**Comment by nils-ohlmeier** [comment]
Besides the SSRC number re-use I commented on this looks like unified SDP to me.
Not sure if or how a plan B implementation would handle this (properly).

**Comment by rwaldron** [comment]
@fippo re: 

>  check that this SDP is what JSEP has for what has been known as "unified plan"?

Can you point me to relevant section to read in https://datatracker.ietf.org/doc/draft-ietf-rtcweb-jsep/?include_text=1? Thanks!

**Comment by fippo** [comment]
https://tools.ietf.org/html/draft-ietf-rtcweb-jsep-20#page-82 ('The SDP for |offer-B2| is shown below') is the closest.

**Comment by rwaldron** [comment]
@fippo 

> https://tools.ietf.org/html/draft-ietf-rtcweb-jsep-20#page-82 ('The SDP for |offer-B2| is shown below') is the closest.

Thank you sir!

**Comment by rwaldron** [comment]
@fippo can you rebase this? Thanks!

**Comment by fippo** [comment]
@rwaldron I won't have time to do so in the foreseeable future. Feel free to go with it.

**Pull request closed without merging**